test/Streamly/Test/Prelude/Serial.hs

```diff
@@ -106,55 +107,160 @@ splitOnSuffixSeq = do
         S.toList
              $ IS.splitOnSuffixSeq (A.fromList pat) (FL.toList) (S.fromList xs)
 
+seqSplitterProperties ::
+       forall a. (Arbitrary a, Eq a, Show a, Storable a, Enum a)
+    => a
+    -> String
+    -> Spec
+seqSplitterProperties sep desc = do
+    describe (desc <> " splitOnSeq")
+        $ do
+
+            forM_ [0, 1, 2, 4] $ intercalateSplitEqId
+            forM_ [0, 1, 2, 4]
+                $ concatSplitIntercalateEqConcat
+                      splitOnSeq_
+                      intercalate
+                      IS.intercalate
```

Comment:
We do not need to pass the last two arguments, we can use them directly. In all these functions.

Reply:
We need to pass intercalateSuffix in case of `splitOnSuffixSeq_`